src/sumcheck.rs

```diff
@@ -0,0 +1,322 @@
+//! Implements sumcheck, which is a system that for some circuit `C`, an input `x` and a witness
+//! `w`, `C(x, w) = 0`.
+
+use crate::{Codec, FieldId, Size};
+use anyhow::{Context, anyhow};
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use educe::Educe;
+use std::{
+    fmt::{self, Formatter},
+    io::{Cursor, Read},
+};
+
+/// A circuit, serialized according to the ad-hoc definition in [1] and [2].
+///
+/// [1]: https://github.com/google/longfellow-zk/blob/87474f308020535e57a778a82394a14106f8be5b/lib/proto/circuit.h
+/// [2]: https://github.com/google/longfellow-zk/blob/87474f308020535e57a778a82394a14106f8be5b/lib/sumcheck/circuit.h
+#[derive(Clone, PartialEq, Eq, Educe)]
+#[educe(Debug)]
+pub struct SerializedCircuit {
+    /// 1 byte version. Currently always 1.
+    version: u8,
+    /// The field this circuit uses. (not clear what subfield is used; hard coded to P256?)
```

Comment:
Since the field used in the example is a prime field, no subfields exist, so presumably we skip encoding any elements in the compressed subfield format. It would be good if there were test vectors exercising subfield-related features as well.